### PR TITLE
php80: 8.0.14 -> 8.0.15

### PR DESCRIPTION
**Pull request by drupol**
PHP 8.0.15 suffer from the same bug from PHP 8.1.2. It has been fixed upstream.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
I think php80Extensions.sockets fails to build now.

@ofborg build php80Extensions.sockets

**Comment by drupol** [comment]
Bug identified: https://github.com/php/php-src/issues/7986

Resolution in progress...

**Comment by drupol** [comment]
@NixOS/ofborg build php80.extensions.sockets
@NixOS/ofborg build php81.extensions.sockets

**Comment by drupol** [comment]
https://github.com/ofborg build php80.extensions.sockets
https://github.com/ofborg build php81.extensions.sockets

File: pkgs/development/interpreters/php/8.0.nix [hunk 1]
```diff
@@ -1,11 +1,13 @@
-{ callPackage, lib, stdenv, ... }@_args:
+{ fetchpatch, callPackage, lib, stdenv, ... }@_args:
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/php/8.0.nix:1]:
```suggestion
{ callPackage, lib, stdenv, ... }@_args:
```

That is unused. Did you try fetching the patch and it did not apply? If so we can remove it again.

Comment (drupol) [reply pkgs/development/interpreters/php/8.0.nix:1]: Yep, I have to remove it. The patch doesn't seems to fix the issue anyway.

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets

File: pkgs/development/interpreters/php/generic.nix [hunk 2]
```diff
@@ -9,6 +9,7 @@ let
     , nixosTests
     , tests
     , fetchurl
+    , fetchpatch
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/php/generic.nix:12]: That is unused now.

File: pkgs/top-level/php-packages.nix [hunk 3]
```diff
@@ -545,11 +545,8 @@ lib.makeScope pkgs.newScope (self: with self; {
         {
           name = "sockets";
           doCheck = false;
-          patches = lib.optional (php.version == "8.1.2")
-            (fetchpatch {
-              url = "https://github.com/php/php-src/commit/07aaa34cd418c44f7bc653fafbf49f07fc71b2bf.patch";
-              sha256 = "sha256-EwVb09/zV2vJ8PuyLpKFCovxe6yKct0UBvishZaordM=";
-            });
+          patches = lib.optional (php.version == "8.1.2" || php.version == "8.0.15")
+            ../development/interpreters/php/Fix-sockets-extension.patch;
```
Comment (SuperSandro2000) [inline pkgs/top-level/php-packages.nix:549]: That file does not exist.

Comment (drupol) [reply pkgs/top-level/php-packages.nix:549]: I know, went for lunch, fixing this asap.

Comment (drupol) [reply pkgs/top-level/php-packages.nix]: Fixed.

File: pkgs/development/interpreters/php/generic.nix (hunk 2 above)
Comment (drupol) [reply pkgs/development/interpreters/php/generic.nix]: Fixed.

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets

File: pkgs/development/interpreters/php/8.0.nix (hunk 1 above)
Comment (drupol) [reply pkgs/development/interpreters/php/8.0.nix]: Fixed.

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets
@GrahamcOfBorg build php80
@GrahamcOfBorg test php80

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets
@GrahamcOfBorg build php80
@GrahamcOfBorg test php80

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets
@GrahamcOfBorg build php80
@GrahamcOfBorg test php80

**Comment by drupol** [comment]
@GrahamcOfBorg build php80.extensions.sockets
@GrahamcOfBorg build php81.extensions.sockets
@GrahamcOfBorg build php80
@GrahamcOfBorg test php80

**Comment by drupol** [comment]
@GrahamcOfBorg build php80

**Comment by drupol** [comment]
@GrahamcOfBorg build php80

**Comment by drupol** [comment]
@GrahamcOfBorg build php80

**Comment by drupol** [comment]
@GrahamcOfBorg build php80

File: pkgs/development/interpreters/php/generic.nix [hunk 4]
```diff
@@ -258,6 +258,7 @@ let
             # Don't record the configure flags since this causes unnecessary
             # runtime dependencies
             ''
+              set -x
```
Comment (ajs124) [inline pkgs/development/interpreters/php/generic.nix:261]: did you intend to commit this?

Comment (drupol) [reply pkgs/development/interpreters/php/generic.nix:261]: No, I'm trying to debug the issue we have with Darwin M1, see: https://github.com/php/php-src/issues/8101

**Comment by drupol** [comment]
@GrahamcOfBorg build php80

**Comment by drupol** [comment]
Superseded by #161038

**Pull request closed without merging**